webpack/assets/javascripts/react_app/components/AutoCompleteSearch/index.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import UUID from 'uuid/v1';
+import { Icon, OverlayTrigger, Tooltip, TypeAheadSelect } from 'patternfly-react';
+import { bindMethods, debounceMethods, noop } from '../../common/helpers';
+import { STATUS } from '../../constants';
+import { TRIGGERS } from './AutoCompleteConstants';
+import './auto-complete.scss';
+
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    bindMethods(this, ['handleClear', 'handleInputChange', 'handleResultsChange', 'handleInputFocus', 'getResults']);
+    debounceMethods(this, 250, ['handleInputChange', 'handleLoading', { name: 'handleInputFocus', time: 110 }]);
```

Comment:
Wondering... Would it be easier to follow as well?

```js
debounceMethods(this, 250, ['handleInputChange', 'handleLoading']);
debounceMethods(this, 110, ['handleInputFocus']);
```

Reply:
I wanted to give an option in `debounceMethods` to control 
each function debounce-time without calling the function couple of times.
e.g:
`debounceMethods(this, 250, [`
    `'handleInputChange',` 
   `'handleLoading',` 
    `{ name: 'handleInputFocus', time: 110 },`
    `{ name: 'handleClear', time: 500 },`
`]);`

Reply:
I can see it useful when each method has different timing, but IMO in your case when you got only one exception, it easier to read it with two calls.